src/github.com/kelseyhightower/confd/backends/etcd/client.go

```diff
@@ -28,10 +30,23 @@ func NewEtcdClient(machines []string, cert, key string, caCert string) (*Client,
 	}
 	// Configure the DialTimeout, since 1 second is often too short
 	c.SetDialTimeout(time.Duration(3) * time.Second)
-	success := c.SetCluster(machines)
-	if !success {
-		return &Client{c}, errors.New("cannot connect to etcd cluster: " + strings.Join(machines, ","))
+
+	maxConnectAttempts := 10
```

Comment:
Yes. What I mean is that the behaviour should stay the same for now, but you should be able to tune (with a flag) how many attempts can be made to connect to etcd before it errors out.

We should probably make this configurable across all backends as well, as this would be a handy feature for more than one backend.